src/infil.c

```diff
@@ -254,6 +279,7 @@ void infil_setState(int j, int m, double x[])
 //  Purpose: sets the current infiltration state for a subcatchment.
 //
 {
+    m = Subcatch[j].infil_type; // 2016-01-15: CHANGE PASS-BY-VALUE, m
```

Comment:
@aerispaha, `m` is an input argument so I think the caller of `infil_setState` should continue to pass in the infiltration type instead of setting it here.